concordium-consensus/src/Concordium/GlobalState/Persistent/Accounts.hs

```diff
@@ -169,34 +264,43 @@ getAccountByCredId cid accs@Accounts{..} =
         Just ai -> fmap (ai,) <$> indexedAccount ai accs
 
 -- | Get the account at a given index (if any).
-getAccountIndex :: (IsProtocolVersion pv, MonadBlobStore m) => AccountAddress -> Accounts pv -> m (Maybe AccountIndex)
-getAccountIndex addr Accounts{..} = AccountMap.lookup addr accountMap
+--  Note that this is looking up via the account alias mechanism introduced in protocol version 3 for all protocol versions.
+--  This is fine as there are no clashes and this approach simplifies the implementation.
```

Comment:
`lookupAccountIndex` will based on the protocol version lookup via an exact match (i.e. when aliases are not supported) and when aliases are supported, then it will lookup via the equivalence class.

I updated documentation for `lookupAccountIndex` to make this clear to any implementer.

Reply:
I changed this, so it preserves semantics for older protocol versions.